demo-client/src/main/java/com/illia/client/config/RestTemplateConfig.java

```diff
@@ -0,0 +1,18 @@
+package com.illia.client.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import org.springframework.web.client.RestTemplate;
+
+
+
+@Configuration
+public class RestTemplateConfig {
+
+    @Bean
+    RestTemplate restTemplate() {
```

Comment:
nit: why this needs a different config class?

Reply:
Primarily for Autowiring RestTemplate (without creating bean manually application fails to start) and for MockBean option, proved being useful in tests